### PR TITLE
feat(firestore): Support array-contains, array-contains-any & in filters

### LINKED ISSUE #2842: Implement the in/array-contains-any queries
Firebase launched yesterday the in/array-contains-any queries. I would like the team implements these requests ASAP : )

Reference:
https://firebase.googleblog.com/2019/11/cloud-firestore-now-supports-in-queries.html
https://firebase.google.com/docs/firestore/query-data/queries

### PR DESCRIPTION
**Pull request by Salakar**
### Summary

Introduce support for newly released `array-contains`, `array-contains-any` & `in` query filters.

**Blog**: https://firebase.googleblog.com/2019/11/cloud-firestore-now-supports-in-queries.html

Closes #2842.

### Checklist

- [x] Supports `Android`
- [x] Supports `iOS`
- [x] `e2e` tests added or updated in packages/**/e2e
- [ ] Flow types updated
- [x] Typescript types updated

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

**Comment by Salakar** [comment]
Looks like the updated BoM is causing Vision tests to fail, might just be the tests are too specific though on checking the returned results.

**Comment by mikehardy** [comment]
> Looks like the updated BoM is causing Vision tests to fail, might just be the tests are too specific though on checking the returned results.

It appears that while the iOS SDK appears to move along as a version-locked monorepo, if you look closely at the underlying version numbers the MLKit for instance is still alpha, so has breaking changes. In camo inside the larger SDK SemVer. I suppose the Android BoM works the same?

https://github.com/firebase/firebase-ios-sdk/issues/3929#issuecomment-540609881

No idea how to handle that

**Comment by Salakar** [comment]
> I suppose the Android BoM works the same?

Aye correct. We'll just have to go fixing internal breaking changes in the code base before release, normally isn't too bad.

**Comment by meftunca** [comment]
How long do we have to wait for the release

**Comment by mikehardy** [comment]
@meftunca I'm sure it would be great if someone could test it, please do so and report back. 

Here I provide a version of the PR you can apply using [patch-package](https://github.com/ds300/patch-package) (which I **swear** by, excellent tool) - 
[@react-native-firebase+firestore+6.0.4.patch.gz](https://github.com/invertase/react-native-firebase/files/3880343/%40react-native-firebase%2Bfirestore%2B6.0.4.patch.gz)

Just install patch-package per instructions (very quick+easy) and unzip that patch into the patches directory. It's a text file, you may visually confirm it contains the functional changes here.

You will need to override the underlying android and ios google SDK versions to do this. I demonstrate the 2 changes you need in your project here to specify the versions that will work: https://github.com/mikehardy/rnfbdemo/blob/master/make-demo-v6.sh#L28

In a few minutes you should have the PR in your project and hearing it works would be very helpful.

@Salakar I haven't figured out how to do patch-package effectively in monorepos yet. The ideal flow is "change yarn dependency in work project from regular version to git, optionally run a local build inside node_modules (typescript etc), run patch-package to generate a diff". In monorepos it's "clone repo, pull branch, run build, copy altered files into node_modules, run patch_package" which is really heavy. I haven't seen anything recommended online on how to solve it other than "[gitpkg](https://github.com/ramasilveyra/gitpkg)" but maybe you have an idea. Either way, I'm playing with v6 now :eyes:

File: tests/ios/Podfile.lock [hunk 1]
```diff
@@ -1,733 +0,0 @@
-PODS:
```
Comment (mikehardy) [inline tests/ios/Podfile.lock]: removing this causes conflicts on rebase to master but it's just a `git add tests/ios/Podfile.lock` fix.

Comment (meftunca) [reply tests/ios/Podfile.lock]: I'm sorry to see you late. You can rest assured that I'll send you feedback after I review it.

**Comment by meftunca** [comment]
> @meftunca I'm sure it would be great if someone could test it, please do so and report back.
> 
> Here I provide a version of the PR you can apply using [patch-package](https://github.com/ds300/patch-package) (which I **swear** by, excellent tool) -
> [@react-native-firebase+firestore+6.0.4.patch.gz](https://github.com/invertase/react-native-firebase/files/3880343/%40react-native-firebase%2Bfirestore%2B6.0.4.patch.gz)
> 
> Just install patch-package per instructions (very quick+easy) and unzip that patch into the patches directory. It's a text file, you may visually confirm it contains the functional changes here.
> 
> You will need to override the underlying android and ios google SDK versions to do this. I demonstrate the 2 changes you need in your project here to specify the versions that will work: https://github.com/mikehardy/rnfbdemo/blob/master/make-demo-v6.sh#L28
> 
> In a few minutes you should have the PR in your project and hearing it works would be very helpful.
> 
 
You can be sure that it works smoothly :)

**Comment by mikehardy** [comment]
Fantastic! Thanks for testing it in the real world, that really helps @meftunca

**Pull request merged**